src/build_systems.rs

```diff
@@ -49,12 +49,23 @@ impl BuildSystems {
                 // TODO: add option to NOT clean
                 println!("Running make fclean");
 
-                let _ = Command::new("make")
+                let command = Command::new("make")
                     .arg("fclean")
                     .envs(shared::DEFAULT_RUN_ENV)
                     .spawn()?
-                    .wait();
-
+                    .wait_with_output();
+
+                if !command.unwrap().status.success() {
+                    println!("Error: Could not run rule 'fclean', trying 'clean'");
+                    let command_fallback = Command::new("make")
```

Comment:
```suggestion
                    let command = Command::new("make")
```

[Variable shadowing](https://doc.rust-lang.org/rust-by-example/variable_bindings/scope.html)